core/src/main/java/org/ehcache/core/spi/store/Store.java

```diff
@@ -334,15 +334,15 @@
    * @param key the key to operate on
    * @param mappingFunction the function that will produce the new value.
    * @param replaceEqual indicates if an equal value replaces the existing one
+   * @param invokeWriter indicates if the writer should be invoked
    * @return the new value associated with the key or {@code null} if none
    *
    * @throws ClassCastException if the specified key is not of the correct type {@code K}
    * @throws NullPointerException if any of the arguments is {@code null}
    * @throws StoreAccessException if the mapping can't be changed
    *
-   * @see #compute(Object, BiFunction)
    */
-  ValueHolder<V> compute(K key, BiFunction<? super K, ? super V, ? extends V> mappingFunction, Supplier<Boolean> replaceEqual) throws StoreAccessException;
+  ValueHolder<V> compute(K key, BiFunction<? super K, ? super V, ? extends V> mappingFunction, Supplier<Boolean> replaceEqual, Supplier<Boolean> invokeWriter) throws StoreAccessException;
```

Comment:
We should probably have two methods here, one with the `invokeWriter` supplier and one without.  This way we can explicitly call out why the invokeWriter one exists (JSR-107... bleagh), and at the same time we can throw an explicit exception from the classes that should be protected from the non false returning supplier invocations by the local loader/writer wrapper.

Reply:
Same goes for other similar places.

Reply:
I tried, but it does not help. The reason being this compute method is only called from cache level, which does not know what kind of store it is dealing with. It will always call the one with invokeWithWriter and that would flow down the hierarchy.

Reply:
I'm confused, there are ton of places where you explicitly call it with `() -> false`.

Reply:
This method has two uses:

- In JSR invoke implementation
- In bulkcompute of the Store

Out of these others are just test cases. Now bulk ones are the internal usage in respective Store's implementaion. The only usage as SPI is in JSR implmentation which makes calls at cache layer.